src/env-inl.h

```diff
@@ -527,6 +554,14 @@ void Environment::SetUnrefImmediate(native_immediate_callback cb,
   CreateImmediate(cb, data, obj, false);
 }
 
+inline bool Environment::can_call_into_js() const {
+  return can_call_into_js_;
```

Comment:
> Is is possible for someone to bump into a pending exception while this returns true?

I would expect that, yes. But the code should be laid out to handle such a situation anyway, right?

The function isn’t meant to give an exhausting result that tells code whether it’s okay to call into JS or not… it’s more of an “allowed to” than “able to”. Maybe we should rename it to `may_call_into_js()`? What do you think?

Reply:
@addaleax In my understanding `may_call_into_js()` is probably not too different from `can_call_into_js()`, maybe add a comment there making it clear that this returning true does not guarantee that one can call into the JS without being careful about pending exceptions? Or, since we only set it to false...maybe `cannot_call_into_js()`?

Reply:
Or, if we don't want to be too board about the meaning of this...maybe just something like `is_in_tear_down()`?

Reply:
> maybe add a comment there making it clear that this returning true does not guarantee that one can call into the JS without being careful about pending exceptions?

Sure, I’m adding a comment. :)

> Or, if we don't want to be too board about the meaning of this...maybe just something like `is_in_tear_down()`?

I think I’d prefer to name it according to how its consumers use it. I don’t know why we would, but I wouldn’t consider it impossible that at some point we may want to use this flag for other things too.